File: quick_sort.js
const fs = require("fs");

function quickSort(arr) {
    if (arr.length <= 1) {
        return arr;
    }

    const pivot = arr[Math.floor(arr.length / 2)];
    const left = arr.filter(x => x < pivot);
    const middle = arr.filter(x => x === pivot);
    const right = arr.filter(x => x > pivot);

    return [...quickSort(left), ...middle, ...quickSort(right)];
}

function getValuesFromDataset(datasetName) {
    const data = fs.readFileSync("datasets/input-" + datasetName + ".txt", "utf8");
    const lines = data.split("\n");

    const numbers = [];
    for (let line of lines) {
        line = line.trim();
        if (line.length > 0) {
            numbers.push(parseInt(line, 10));
        }
    }

    return numbers;
}

function main() {
    if (process.argv.length !== 3) {
        console.log("Usage: node quick_sort.js <dataset_size>");
        process.exit(1);
    }

    const datasetName = process.argv[2];
    const values = getValuesFromDataset(datasetName);
    console.log(quickSort(values));
}

main();

File: quick_sort.py
import sys

def quick_sort(arr):
    if len(arr) <= 1:
        return arr

    pivot = arr[len(arr) // 2]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]

    return quick_sort(left) + middle + quick_sort(right)

def get_values_from_dataset(dataset_size):
    numbers = []
    with open(f"datasets/input-{dataset_size}.txt", "r") as file:
        for line in file:
            line = line.strip()
            if line:
                numbers.append(int(line))
    return numbers

def main():
    if len(sys.argv) != 2:
        print("Usage: python quick_sort.py <dataset>")
        sys.exit(1)

    dataset_size = sys.argv[1]
    values = get_values_from_dataset(dataset_size)
    print(quick_sort(values))

if __name__ == "__main__":
    main()
